File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln39LogInjection.java
package com.example.vulnapp.vulns;

import org.springframework.web.bind.annotation.*;

@RestController
public class Vuln39LogInjection {
    @GetMapping("/vuln39/note")
    public String note(@RequestParam String text) {
        // VULN:VULN-39:CWE-117:taint user input logged without neutralising CR/LF (forged log lines)
        System.out.println("[audit] user note: " + text);
        return "{\"logged\":true}";
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln39LogInjection.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*


@RestController
class Vuln39LogInjection {

    @GetMapping("/vuln39/note")
    fun note(@RequestParam text: String): String {
        // VULN:VULN-39:CWE-117:taint user input logged without neutralising CR/LF
        println("[audit] user note: $text")
        return "{\"logged\":true}"
    }
}
